src/main/scala/components/memory/base-memory-components.scala

```diff
@@ -8,40 +8,51 @@ import chisel3.util.experimental.loadMemoryFromFile
 /**
   * Base class for all modular backing memories. Simply declares the IO and the memory file.
   */
-class BaseDualPortedMemory(size: Int, memfile: String) extends Module {
+abstract class BaseDualPortedMemory(size: Int, memfile: String) extends Module {
   def wireMemory (imem: BaseIMemPort, dmem: BaseDMemPort): Unit = {
     // Connect memory imem IO to dmem accessor
-    this.io.imem.request <> imem.io.request
-    imem.io.response <> this.io.imem.response
+    this.io.imem.request <> imem.io.bus.request
```

Comment:
Maybe this should be in a separate patch? It's not clear how this relates to the CPU design.

Reply:
It shouldn't be appearing in the pull request, because these were changes that were already merged in from the previous PR. 

In hindsight I probably should have made a new branch to submit a pull request, but since this CPU design was an "extension" from the previous pull request. I thought it would be possible to push my new changes onto the same branch and have GitHub display just my new changes.